ethcore/src/client/client.rs

```diff
@@ -2191,6 +2192,13 @@ impl IoClient for Client {
 				let first = queued.write().1.pop_front();
 				if let Some((unverified, receipts_bytes)) = first {
 					let hash = unverified.hash();
+					let parent_hash = unverified.parent_hash();
+					let status = client.importer.block_queue.status(&parent_hash);
```

Comment:
This queue is for new blocks. Ancient blocks are queued in `self.queued_ancient_blocks`. Could check `queued.read().0.contains(&parent_hash);` instead.

Reply:
This checks whether the parent hash is in `importer`. The issue is that above when we report parent status, the block can be in the `importer`, thus break the assumption we have for ancient queues. I don't think it has anything to do with ancient block queues.